@xen-orchestra/lite/src/libs/utils.ts

```diff
@@ -175,3 +180,12 @@ export function parseRamUsage(
 
 export const getFirst = <T>(value: T | T[]): T | undefined =>
   Array.isArray(value) ? value[0] : value;
+
+export const isOperationsPending = (
+  obj: XenApiVm,
+  operations: string[] | string
+) => {
+  const objOperations = Object.values(obj.current_operations);
+  const _operations = Array.isArray(operations) ? operations : [operations];
+  return difference(_operations, objOperations).length < _operations.length;
+};
```

Comment:
I had a hard time to "decode" this function, especially the last line.

If I understand correctly, we want to check whether passed `operations` is/are present in `obj.current_operation`, right?

If so, here is my suggestion:
```suggestion
export const isOperationsPending = (
  obj: XenApiVm,
  operations: string[] | string
) => {
  const currentOperations = Object.values(obj.current_operations);
  return castArray(operations).some((operation) =>
    currentOperations.includes(operation)
  );
};
```